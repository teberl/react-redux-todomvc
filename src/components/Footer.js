import React from 'react';
import PropTypes from 'prop-types';
import FilterLink from './FilterLink';
import {
  SHOW_ALL,
  SHOW_COMPLETED,
  SHOW_ACTIVE
} from '../constants/todoFilters';

const FILTER_TITLES = {
  [SHOW_ALL]: 'All',
  [SHOW_ACTIVE]: 'Active',
  [SHOW_COMPLETED]: 'Completed'
};

Footer.propTypes = {
  completedCount: PropTypes.number.isRequired,
  activeCount: PropTypes.number.isRequired,
  onClearCompleted: PropTypes.func.isRequired,
  filter: PropTypes.oneOf([SHOW_ALL, SHOW_ACTIVE, SHOW_COMPLETED])
};

function Footer(props) {
  const {
    activeCount,
    completedCount,
    onClearCompleted,
    filter: selectedFilter
  } = props;
  const itemWord = activeCount === 1 ? 'item' : 'items';
  const isActive = filter => filter === selectedFilter;

  return (
    <footer className="footer">
      <span className="todo-count">
        <strong>{activeCount || 'No'}</strong> {itemWord} left
      </span>
      <ul className="filters">
        {Object.keys(FILTER_TITLES).map(filter => (
          <li key={filter}>
            <FilterLink active={isActive(filter)} filter={filter}>
              {FILTER_TITLES[filter]}
            </FilterLink>
          </li>
        ))}
      </ul>
      {!!completedCount && (
        <button className="clear-completed" onClick={onClearCompleted}>
          Clear completed
        </button>
      )}
    </footer>
  );
}

export default Footer;
