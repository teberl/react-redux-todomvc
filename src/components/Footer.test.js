import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter as Router, Route } from 'react-router-dom';
import Footer from './Footer';
import { SHOW_ALL, SHOW_ACTIVE } from '../constants/todoFilters';

describe('Footer', () => {
  describe('with SHOW_ALL filter', () => {
    test('renders the items count and clear button correct', () => {
      const footerProps = {
        activeCount: 1,
        completedCount: 1,
        onClearCompleted: jest.fn(),
        filter: SHOW_ALL
      };
      const testRenderer = TestRenderer.create(
        <Router>
          <Route render={props => <Footer {...footerProps} {...props} />} />
        </Router>
      ).toJSON();

      expect(testRenderer).toMatchSnapshot();
    });

    test('renders the items count correct and no clear button', () => {
      const footerProps = {
        activeCount: 0,
        completedCount: 0,
        onClearCompleted: jest.fn(),
        filter: SHOW_ALL
      };
      const testRenderer = TestRenderer.create(
        <Router>
          <Route render={props => <Footer {...footerProps} {...props} />} />
        </Router>
      ).toJSON();

      expect(testRenderer).toMatchSnapshot();
    });
  });
});

describe('with SHOW_ACTIVE filter', () => {
  test('renders the items count and clear button correct', () => {
    const footerProps = {
      activeCount: 4,
      completedCount: 1,
      onClearCompleted: jest.fn(),
      filter: SHOW_ACTIVE
    };
    const testRenderer = TestRenderer.create(
      <Router>
        <Route render={props => <Footer {...footerProps} {...props} />} />
      </Router>
    ).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });
});
