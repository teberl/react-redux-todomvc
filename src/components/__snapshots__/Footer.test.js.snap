// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer with SHOW_ALL filter renders the items count and clear button correct 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      1
    </strong>
     
    item
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className="selected"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className=""
        href="/show_active"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className=""
        href="/show_completed"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={[MockFunction]}
  >
    Clear completed
  </button>
</footer>
`;

exports[`Footer with SHOW_ALL filter renders the items count correct and no clear button 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      No
    </strong>
     
    items
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className="selected"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className=""
        href="/show_active"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className=""
        href="/show_completed"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
</footer>
`;

exports[`with SHOW_ACTIVE filter renders the items count and clear button correct 1`] = `
<footer
  className="footer"
>
  <span
    className="todo-count"
  >
    <strong>
      4
    </strong>
     
    items
     left
  </span>
  <ul
    className="filters"
  >
    <li>
      <a
        className=""
        href="/"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        All
      </a>
    </li>
    <li>
      <a
        className="selected"
        href="/show_active"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Active
      </a>
    </li>
    <li>
      <a
        className=""
        href="/show_completed"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Completed
      </a>
    </li>
  </ul>
  <button
    className="clear-completed"
    onClick={[MockFunction]}
  >
    Clear completed
  </button>
</footer>
`;
